integration-provisioning/src/main/java/org/symphonyoss/integration/provisioning/service/ApplicationService.java

```diff
@@ -116,10 +125,28 @@ public void setupApplication(IntegrationSettings settings, Application applicati
       AppStoreWrapper wrapper =
           AppStoreBuilder.build(application, domain, settings.getConfigurationId(), botUserId);
 
-      Map<String, String> app = client.getAppByAppGroupId(appType, DEFAULT_USER_ID);
+      Map<String, Object> app = client.getAppByAppGroupId(appType, DEFAULT_USER_ID);
 
       if (app != null) {
-        wrapper.setId(app.get(APP_ID));
+        if (app.get(APP_ID) != null) {
+          wrapper.setId((String) app.get(APP_ID));
+        }
+
+        if (existsAppSettings(app)) {
+          HashMap<String, Object> appSettings = ((HashMap<String, Object>) app.get(APP_SETTINGS));
```

Comment:
This code inside this if reapeats twice. Shouldn't be better to put it in one place?

Reply:
Done.